### DISCUSSION TITLE
using-serilog-aspnetcore-in-asp-net-core-3-logging-the-selected-endpoint-name-with-serilog/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#56] [Announcements]
# using-serilog-aspnetcore-in-asp-net-core-3-logging-the-selected-endpoint-name-with-serilog/

In this post I show how to add extra details to Serilog.AspNetCore's request logging middleware, such as the host name or the selected Endpoint name.

https://andrewlock.net/using-serilog-aspnetcore-in-asp-net-core-3-logging-the-selected-endpoint-name-with-serilog/

**Comment by ahmedanwar100** [comment]
I want to log requests body too, so how I can achieve it

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at January 07 2020, 06:48, in reply to [AlexandreJobin](https://disqus.com/by/AlexandreJobin/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/56#discussioncomment-6058045)</em>

---

<p>As discussed in that issue, both ways are good :) Personally I would tend towards the middleware approach for request-scoped properties, and the Enricher approach for e.g. machine or process-level properties</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at January 08 2020, 10:29, in reply to [Paulo Morgado](https://disqus.com/by/paulo_morgado/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/56#discussioncomment-6058050)</em>

---

<p>ASP.NET Core! :) <a href="https://devblogs.microsoft.com/aspnet/improvements-in-net-core-3-0-for-troubleshooting-and-monitoring-distributed-apps/" rel="nofollow noopener" title="https://devblogs.microsoft.com/aspnet/improvements-in-net-core-3-0-for-troubleshooting-and-monitoring-distributed-apps/">https://devblogs.microsoft.com/aspnet/improvements-in-net-core-3-0-for-troubleshooting-and-monitoring-distributed-apps/</a></p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at February 21 2020, 05:34, in reply to [Rob Nelder](https://disqus.com/by/robnelder/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/56#discussioncomment-6058058)</em>

---

<p>If you're using ASP.NET Core 3.0 functionality, you can't target .NET Standard 2.0, you have to target .NET Core 3.x and use a &lt;targetframework&gt; element. Check out this post: <a href="https://andrewlock.net/converting-a-netstandard-2-library-to-netcore-3/#libraries-that-depend-on-asp-net-core-nuget-packages" rel="nofollow noopener" title="https://andrewlock.net/converting-a-netstandard-2-library-to-netcore-3/#libraries-that-depend-on-asp-net-core-nuget-packages">https://andrewlock.net/converting-a-netstandard-2-library-to-netcore-3/#libraries-that-depend-on-asp-net-core-nuget-packages</a></p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 04 2020, 08:43, in reply to [Pavel](https://disqus.com/by/disqus_RNRmwBlrG3/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/56#discussioncomment-6058065)</em>

---

<p>No, it's a Serilog concept, but the console and file sinks only write some of the information. You can customise what they write, but they're generally not designed for structured logging like Seq and Elasticsearch are.</p>

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 04 2020, 10:35, in reply to [Pavel](https://disqus.com/by/disqus_RNRmwBlrG3/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/56#discussioncomment-6058071)</em>

---

<p>You can customize the template: <a href="https://github.com/serilog/serilog-sinks-console#output-templates" rel="nofollow noopener" title="https://github.com/serilog/serilog-sinks-console#output-templates">https://github.com/serilog/serilog-sinks-console#output-templates</a></p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 17 2020, 06:53, in reply to [Faisal Fayyaz](https://disqus.com/by/disqus_BvV1sCDBtX/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/56#discussioncomment-6058075)</em>

---

<p>You can customize the console template as described here: <a href="https://github.com/serilog/serilog-sinks-console#output-templates" rel="nofollow noopener" title="https://github.com/serilog/serilog-sinks-console#output-templates">https://github.com/serilog/serilog-sinks-console#output-templates</a></p>

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 17 2020, 07:12, in reply to [Faisal Fayyaz](https://disqus.com/by/disqus_BvV1sCDBtX/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/56#discussioncomment-6058077)</em>

---

<p>Correct, the Console and File sinks do not display these values by default. They will be included with structural loggers like Seq automatically.</p>

[a comment by an automated account is not shown]
